p2panda-rs/src/operation/verified_operation.rs

```diff
@@ -2,178 +2,84 @@
 
 use std::hash::Hash as StdHash;
 
-use crate::entry::{decode_entry, EntrySigned};
 use crate::identity::Author;
-use crate::operation::{AsVerifiedOperation, Operation, OperationEncoded, VerifiedOperationError};
-use crate::Validate;
-
-use super::OperationId;
+use crate::operation::traits::AsVerifiedOperation;
+use crate::operation::{Operation, OperationId};
 
 /// An operation which has been encoded and published on a signed entry.
 ///
 /// Contains the values of an operation as well as its author and id. This
 /// [operation id][OperationId] is only available on [`VerifiedOperation`] and not on
 /// [`Operation`] because it is derived from the hash of the signed entry an operation is encoded
 /// on.
-#[derive(Debug, Clone, Eq, PartialEq, StdHash)]
+#[derive(Debug, Clone)]
 pub struct VerifiedOperation {
-    /// The hash of the entry this operation was published with.
-    operation_id: OperationId,
+    /// Identifier of the operation.
+    pub(crate) operation_id: OperationId,
 
-    /// The public key of the author who published this operation.
-    public_key: Author,
+    /// Operation, which is the payload of the entry.
+    pub(crate) operation: Operation,
 
-    /// The actual operation this struct wraps.
-    operation: Operation,
+    /// The public key of the key pair used to publish this operation.
+    pub(crate) public_key: Author,
 }
 
 impl AsVerifiedOperation for VerifiedOperation {
-    type VerifiedOperationError = VerifiedOperationError;
-
-    /// Returns a new `VerifiedOperation` instance.
-    ///
-    /// Use `VerifiedOperation::new_from_entry()` instead if you want to validate that the operation
-    /// was signed by this public key.
-    fn new(
```

Comment:
It's only possible to get a `VerifiedOperation` via `validate_operation_with_entry()` which requires a whole bunch (schema, entry etc...) however when we get operations from the `OperationStore` they come out as a load of strings and need to be parsed into a `VerifiedOperation`. Any thoughts on how we might do this nicely @cafca & @adzialocha ?

Reply:
It sounds a little bit like you need a different data type in aquadoggo, something which got verified in the past but that's it, from that point you don't need it anymore. Maybe we introduce a new struct in the doggo which just represents what you have in the database?

Reply:
Isn't it funny how the separation of entry and operation and it's dependency creeps back at us all the time? 😅 I can see how it is fine to combine objects without checking but I'm not sure yet if the core library needs to provide such a thing?

Reply:
> It sounds a little bit like you need a different data type in aquadoggo, something which got verified once in the beginning. Verifying it a second time doesn't make sense after you already stored it in the database. Maybe we introduce a new struct in the doggo which just represents what you have in the database? Like a simple struct which just holds the required fields?

yeh, I'm considering this, it's actually how things were earlier in the storage provider story, but the structs got merged at some point. Slight worry here is that we may still need a `VerifiedOperation` at some point, but this is what we have the traits for, so it could hopefully work out.